spec/classes/init_spec.rb

```diff
@@ -268,6 +269,27 @@
     end
   end
 
+  describe 'with \'fqdn_ip_address\' parameter set to' do
+    context 'an invalid type (not ip address)' do
+      let(:params) { 
+        { :fqdn_ip_address => 'wrong.ip.address',
+        } 
+      }
+      let(:facts) {
+        { :hostname  => 'monkey',
+          :ipaddress => '10.1.2.3',
+          :fqdn      => 'monkey.example.com',
+        }
+      }
+
+      it 'should fail' do
+        expect {
+          should contain_class('hosts')
+        }.to raise_error(Puppet::Error,/hosts::fqdn_ip_address must be an IP address./)
```

Comment:
The period at the end is actually a matcher similar to `*` and should be escaped

``` ruby
}.to raise_error(Puppet::Error,/hosts::fqdn_ip_address must be an IP address\./)
```